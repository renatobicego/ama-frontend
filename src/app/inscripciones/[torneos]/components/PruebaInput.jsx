import { Input, Option, Select } from "@/app/utils/materialTailwind"

const PruebaInput = ({pruebas, pruebaAgregada, setPruebasSelected}) => {


    const handleSelect = (value) => {
        setPruebasSelected(prevState => prevState.map(p => {
            if(p.id === pruebaAgregada.id){
                return {
                    ...p,
                    prueba: value
                }
            }else{
                return p
            }
        }))
    }

    const handleInputMarca = (event) => {
        setPruebasSelected(prevState => prevState.map(p => {
            if(p.id === pruebaAgregada.id){
                return {
                    ...p,
                    marca: event.target.value
                }
            }else{
                return p
            }
        }))
    }

    return (
        <div className="flex w-full justify-between gap-6">
            <Select value={pruebaAgregada.value} onChange={handleSelect} color="gray" label="Seleccionar Prueba*">
                {pruebas.map((p, i) => 
                    <Option key={i} value={p.value}>{p.name}</Option>
                )}
            </Select>
            <Input value={pruebaAgregada.marca} onChange={handleInputMarca} color="gray" label="Mejor Marca de Prueba*" />
            
        </div>
    )
}

export default PruebaInput