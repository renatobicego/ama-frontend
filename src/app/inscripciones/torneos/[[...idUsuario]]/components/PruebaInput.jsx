import {
  Input,
  Option,
  Select,
  Typography,
} from "@/app/utils/materialTailwind";
import { setFormatoMarca } from "@/app/utils/utils";
import { InformationCircleIcon, TrashIcon } from "@heroicons/react/24/outline";
import axios from "axios";
import { validate } from "uuid";
import { PatternFormat } from "react-number-format";

const PruebaInput = ({
  pruebas,
  pruebaAgregada,
  setPruebasSelected,
  editando,
}) => {
  const handleSelect = (value) => {
    const pruebaSelected = pruebas.find((p) => p._id === value);
    setPruebasSelected((prevState) =>
      prevState.map((p) => {
        // Change only pruebaAgregada input
        if (p.id === pruebaAgregada.id) {
          return {
            ...p,
            prueba: value,
            formato: setFormatoMarca(pruebaSelected),
          };
        } else {
          return p;
        }
      })
    );
  };

  const handleInputMarca = (event) => {
    setPruebasSelected((prevState) =>
      prevState.map((p) => {
        // Change only pruebaAgregada input
        if (p.id === pruebaAgregada.id) {
          return {
            ...p,
            marca: event.target.value.trim(),
          };
        } else {
          return p;
        }
      })
    );
  };

  const deletePrueba = async () => {
    setPruebasSelected((prevState) =>
      prevState.filter((p) => p.id !== pruebaAgregada.id)
    );
    if (editando && !validate(pruebaAgregada.id)) {
      await axios.delete(
        `${process.env.NEXT_PUBLIC_URL_API}/pruebas_atleta/${pruebaAgregada.id}`
      );
    }
  };

  return (
    <div className="flex w-full justify-between gap-6 flex-wrap md:flex-nowrap relative lg:flex-col">
      {/* <Select 
                value={pruebaAgregada.prueba} 
                onChange={handleSelect} 
                color="gray" 
                label="Seleccionar Prueba*"
                aria-labelledby="prueba"
                labelProps={{id: 'prueba'}}
                >
                    
                {pruebas.map(p => 
                    <Option key={p._id} value={p._id}>{p.nombre}</Option>
                )}
            </Select> */}
      <div className="flex flex-col">
        <label>Seleccionar Prueba</label>
        <select
          onChange={(event) => handleSelect(event.target.value)}
          value={pruebaAgregada.prueba}
          placeholder="Prueba*"
          className={`border border-gray-400 py-2.5 px-2 text-sm rounded-md`}
        >
          <option key={"disabled"} hidden>
            Seleccionar Prueba
          </option>
          {pruebas.map((p) => (
            <option key={p._id} value={p._id}>
              {p.nombre}
            </option>
          ))}
        </select>
      </div>
      {pruebaAgregada.formato && (
        <div className="w-full">
          {/* <Input 
                        value={pruebaAgregada.marca ? pruebaAgregada.marca : ''} 
                        onChange={handleInputMarca} 
                        color="gray" 
                        label="Mejor Marca de Prueba*"
                        aria-labelledby="marca-prueba"
                        labelProps={{id: 'marca-prueba'}} 
                        /> */}
          <PatternFormat
            value={pruebaAgregada.marca ? pruebaAgregada.marca : ""}
            onChange={handleInputMarca} 
            format={pruebaAgregada.formato}
            className="border-2 border-gray-400"
            placeholder="Marca"
          />
          <Typography
            variant="small"
            color="gray"
            className="flex items-center gap-1 font-normal mt-2 w-full"
          >
            <InformationCircleIcon className="w-4 h-4 -mt-px" />
            El formato de la marca debe ser {pruebaAgregada.formato.replaceAll('#', '0')} . AGREGAR LOS CEROS CORRESPONDIENTES. 
            Por ejemplo, si es una prueba de
            fondo, la marca 08:18:35 serían 8 minutos, 18 segundos y 35 centésimas.
          </Typography>
        </div>
      )}

      <button
        type="button"
        aria-label="delete prueba"
        className={`absolute -right-6 top-[42%] md:-right-10 md:top-2 rounded cursor-pointer`}
        onClick={deletePrueba}
      >
        <TrashIcon
          strokeWidth={1}
          className="h-5 w-5 md:h-7 md:w-7 text-gray-700"
        />
      </button>
    </div>
  );
};

export default PruebaInput;
