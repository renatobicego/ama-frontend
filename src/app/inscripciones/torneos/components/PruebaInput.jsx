import { Input, Option, Select } from "@/app/utils/materialTailwind"
import { TrashIcon } from "@heroicons/react/24/outline"

const PruebaInput = ({pruebas, pruebaAgregada, setPruebasSelected}) => {


    const handleSelect = (value) => {
        setPruebasSelected(prevState => prevState.map(p => {
            if(p.id === pruebaAgregada.id){
                return {
                    ...p,
                    prueba: value
                }
            }else{
                return p
            }
        }))
    }

    const handleInputMarca = (event) => {
        setPruebasSelected(prevState => prevState.map(p => {
            if(p.id === pruebaAgregada.id){
                return {
                    ...p,
                    marca: event.target.value
                }
            }else{
                return p
            }
        }))
    }

    const deletePrueba = () => {
        setPruebasSelected(prevState => prevState.filter(p => p.id !== pruebaAgregada.id))
    }

    return (
        <div className="flex w-full justify-between gap-6 flex-wrap md:flex-nowrap relative">
            <Select 
                value={pruebaAgregada.value} 
                onChange={handleSelect} 
                color="gray" 
                label="Seleccionar Prueba*">
                    
                {pruebas.map((p, i) => 
                    <Option key={i} value={p.value}>{p.name}</Option>
                )}
            </Select>
            <Input value={pruebaAgregada.marca} onChange={handleInputMarca} color="gray" label="Mejor Marca de Prueba*" />

            <button 
                className={`absolute -right-6 top-[42%] md:-right-10 md:top-2 rounded cursor-pointer`}
                onClick={deletePrueba}
            >
                <TrashIcon strokeWidth={1} className="h-5 w-5 md:h-7 md:w-7 text-gray-700" />
            </button>
        </div>
    )
}

export default PruebaInput