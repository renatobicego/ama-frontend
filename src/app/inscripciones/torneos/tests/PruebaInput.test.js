import '@testing-library/jest-dom/extend-expect'
import { cleanup, fireEvent, render, screen } from '@testing-library/react'
import PruebaInput from '../components/PruebaInput'

describe('<PruebaInput />', () => {

    let pruebaAgregada
    let pruebaInput
    let setPruebasSelected

    beforeEach(() => {
        pruebaAgregada = {
            id: '',
            prueba: '',
            marca: ''
        }
        setPruebasSelected = jest.fn()
        pruebaInput = render(
            <PruebaInput 
                pruebas={[]} 
                setPruebasSelected={setPruebasSelected}
                pruebaAgregada={pruebaAgregada}
            />)
    })

    afterEach(() => {
        cleanup()
    })


    test('renders content', () => {
        expect(pruebaInput.container).toMatchSnapshot()

        // Renders select prueba
        const selectPrueba = screen.getByLabelText('Seleccionar Prueba*')
        expect(selectPrueba).toBeInTheDocument()

        // Renders input maca
        const marcaPrueba = screen.getByLabelText('Mejor Marca de Prueba*')
        expect(marcaPrueba).toBeInTheDocument()

        // Renders delete Prueba button
        const deleteButton = screen.getByRole('button', { name: 'delete prueba' });
        expect(deleteButton).toBeInTheDocument()
    })

    test('setPruebasSelected is called when delete button is clicked', () => {

        const deleteButton = screen.getByRole('button', { name: 'delete prueba' });

        fireEvent.click(deleteButton)

        expect(setPruebasSelected).toHaveBeenCalledTimes(1)

    })

    test('change input marca', () => {
        // Get marca input
        const marcaPrueba = screen.getByLabelText('Mejor Marca de Prueba*')

        // Change input
        fireEvent.change(marcaPrueba, { target: { value: 'marca test' } })
        
        expect(setPruebasSelected).toHaveBeenCalledTimes(1)
        
    })  

    test('change select prueba', () => {
        // Get pais input
        const selectPrueba = screen.getByLabelText('Seleccionar Prueba*')

        // Change input
        fireEvent.change(selectPrueba, { target: { value: 'jabalina' } })
        
        expect(selectPrueba.value).toBe('jabalina')
        
    })  

    
})