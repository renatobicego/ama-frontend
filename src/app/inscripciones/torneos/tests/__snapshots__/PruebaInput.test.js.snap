// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PruebaInput /> renders content 1`] = `
<div>
  <div
    class="flex w-full justify-between gap-6 flex-wrap md:flex-nowrap relative"
  >
    <div
      class="relative w-full min-w-[200px] h-10"
    >
      <button
        aria-expanded="false"
        aria-haspopup="listbox"
        aria-labelledby="prueba"
        class="peer w-full h-full bg-transparent text-blue-gray-700 font-sans font-normal text-left outline outline-0 focus:outline-0 disabled:bg-blue-gray-50 disabled:border-0 transition-all border text-sm px-3 py-2.5 rounded-[7px] border-blue-gray-200"
        role="combobox"
        type="button"
      >
        <span
          class="absolute top-2/4 -translate-y-2/4 left-3 pt-0.5"
        />
        <div
          class="grid place-items-center absolute top-2/4 right-2 pt-px w-5 h-5 text-blue-gray-400 rotate-0 -translate-y-2/4 transition-all"
        >
          <svg
            fill="currentColor"
            viewBox="0 0 20 20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M5.293 7.293a1 1 0 011.414 0L10 10.586l3.293-3.293a1 1 0 111.414 1.414l-4 4a1 1 0 01-1.414 0l-4-4a1 1 0 010-1.414z"
              fill-rule="evenodd"
            />
          </svg>
        </div>
      </button>
      <label
        class="flex w-full h-full select-none pointer-events-none absolute left-0 font-normal transition-all -top-1.5 before:content[' '] before:block before:box-border before:w-2.5 before:h-1.5 before:mt-[6.5px] before:mr-1 before:rounded-tl-md before:pointer-events-none before:transition-all peer-disabled:before:border-transparent after:content[' '] after:block after:flex-grow after:box-border after:w-2.5 after:h-1.5 after:mt-[6.5px] after:ml-1 after:rounded-tr-md after:pointer-events-none after:transition-all peer-disabled:after:border-transparent text-sm peer-disabled:text-blue-gray-400 leading-[3.75] text-blue-gray-400 before:border-transparent after:border-transparent"
        id="prueba"
      >
        Seleccionar Prueba*
      </label>
    </div>
    <div
      class="relative w-full min-w-[200px] h-10"
    >
      <input
        aria-labelledby="marca-prueba"
        class="peer w-full h-full bg-transparent text-blue-gray-700 font-sans font-normal outline outline-0 focus:outline-0 disabled:bg-blue-gray-50 disabled:border-0 transition-all placeholder-shown:border placeholder-shown:border-blue-gray-200 placeholder-shown:border-t-blue-gray-200 border focus:border-2 border-t-transparent focus:border-t-transparent text-sm px-3 py-2.5 rounded-[7px] border-blue-gray-200 focus:border-gray-500"
        placeholder=" "
        value=""
      />
      <label
        class="flex w-full h-full select-none pointer-events-none absolute left-0 font-normal peer-placeholder-shown:text-blue-gray-500 leading-tight peer-focus:leading-tight peer-disabled:text-transparent peer-disabled:peer-placeholder-shown:text-blue-gray-500 transition-all -top-1.5 peer-placeholder-shown:text-sm text-[11px] peer-focus:text-[11px] before:content[' '] before:block before:box-border before:w-2.5 before:h-1.5 before:mt-[6.5px] before:mr-1 peer-placeholder-shown:before:border-transparent before:rounded-tl-md before:border-t peer-focus:before:border-t-2 before:border-l peer-focus:before:border-l-2 before:pointer-events-none before:transition-all peer-disabled:before:border-transparent after:content[' '] after:block after:flex-grow after:box-border after:w-2.5 after:h-1.5 after:mt-[6.5px] after:ml-1 peer-placeholder-shown:after:border-transparent after:rounded-tr-md after:border-t peer-focus:after:border-t-2 after:border-r peer-focus:after:border-r-2 after:pointer-events-none after:transition-all peer-disabled:after:border-transparent peer-placeholder-shown:leading-[3.75] text-blue-gray-400 peer-focus:text-gray-500 before:border-blue-gray-200 peer-focus:before:!border-gray-500 after:border-blue-gray-200 peer-focus:after:!border-gray-500"
        id="marca-prueba"
      >
        Mejor Marca de Prueba*
         
      </label>
    </div>
    <button
      aria-label="delete prueba"
      class="absolute -right-6 top-[42%] md:-right-10 md:top-2 rounded cursor-pointer"
    >
      <svg
        aria-hidden="true"
        class="h-5 w-5 md:h-7 md:w-7 text-gray-700"
        fill="none"
        stroke="currentColor"
        stroke-width="1"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M14.74 9l-.346 9m-4.788 0L9.26 9m9.968-3.21c.342.052.682.107 1.022.166m-1.022-.165L18.16 19.673a2.25 2.25 0 01-2.244 2.077H8.084a2.25 2.25 0 01-2.244-2.077L4.772 5.79m14.456 0a48.108 48.108 0 00-3.478-.397m-12 .562c.34-.059.68-.114 1.022-.165m0 0a48.11 48.11 0 013.478-.397m7.5 0v-.916c0-1.18-.91-2.164-2.09-2.201a51.964 51.964 0 00-3.32 0c-1.18.037-2.09 1.022-2.09 2.201v.916m7.5 0a48.667 48.667 0 00-7.5 0"
          stroke-linecap="round"
          stroke-linejoin="round"
        />
      </svg>
    </button>
  </div>
</div>
`;
